src/platform/bouffalolab/BL602/DiagnosticDataProviderImpl.cpp

```diff
@@ -163,11 +163,33 @@ CHIP_ERROR DiagnosticDataProviderImpl::GetBootReason(BootReasonType & bootReason
     return CHIP_NO_ERROR;
 }
 
+static int bl_netif_get_all_ip6(struct netif * netif, ip6_addr_t if_ip6[])
+{
+    if (netif == NULL || if_ip6 == NULL)
+    {
+        return 0;
+    }
+
+    int addr_count = 0;
+    for (int i = 0; i < LWIP_IPV6_NUM_ADDRESSES; i++)
+    {
+        if (!ip_addr_cmp(&netif->ip6_addr[i], IP6_ADDR_ANY))
+        {
+            memcpy(&if_ip6[addr_count++], &netif->ip6_addr[i], sizeof(ip6_addr_t));
+        }
+    }
+
+    return addr_count;
+}
+
 CHIP_ERROR DiagnosticDataProviderImpl::GetNetworkInterfaces(NetworkInterface ** netifpp)
 {
     NetworkInterface * ifp = new NetworkInterface();
     struct netif * netif;
     uint8_t mac_addr[6];
+    uint32_t ip, gw, mask;
+    ip6_addr_t ip6_addr[kMaxIPv6AddrCount];
+    uint8_t ipv6_addr_count = 0;
```

Comment:
this is C++ - you can declare variables where they are used instead of at the top of the function, to make code easier to read.